Userland/Libraries/LibGUI/Application.cpp

```diff
@@ -324,4 +325,73 @@ void Application::event(Core::Event& event)
     Object::event(event);
 }
 
+void Application::set_config_domain(String config_domain)
+{
+    m_config_domain = move(config_domain);
+}
+
+void Application::register_recent_file_actions(Badge<GUI::Menu>, Vector<NonnullRefPtr<GUI::Action>> actions)
+{
+    m_recent_file_actions = move(actions);
+    update_recent_file_actions();
+}
+
+void Application::update_recent_file_actions()
```

Comment:
I think adding a `VERIFY(!m_config_domain.is_empty())` would be useful here, as currently we would pass the empty domain string to ConfigServer if you forgot to call `Application::set_config_domain()`.